eslint-plugin-prettier.js

```diff
@@ -397,19 +398,58 @@ module.exports = {
               context.options[0] === 'fb'
                 ? FB_PRETTIER_OPTIONS
                 : context.options[0];
+
             const prettierRcOptions =
               usePrettierrc &&
               prettier.resolveConfig &&
               prettier.resolveConfig.sync
-                ? prettier.resolveConfig.sync(context.getFilename(), {
+                ? prettier.resolveConfig.sync(filepath, {
                     editorconfig: true
                   })
                 : null;
+
+            // prettier.getFileInfo was added in v1.13
+            const prettierFileInfo =
+              prettier.getFileInfo && prettier.getFileInfo.sync
+                ? prettier.getFileInfo.sync(filepath, {
+                    ignorePath: '.prettierignore'
+                  })
+                : { ignored: false, inferredParser: null };
+
+            // Skip if file is ignored using a .prettierignore file
+            if (prettierFileInfo.ignored) {
+              return;
+            }
+
+            const initialOptions = {};
+
+            // Eslint suppports processors that let you extract and lint JS
+            // fragments within a non-JS language. In the cases where prettier
+            // supports the same language as a processor, we want to process
+            // the provided source code as javascript (as eslint provides the
+            // rules with fragments of JS) instead of guessing the parser
+            // based off the filename. Otherwise, for instance, on a .md file we
+            // end up trying to run prettier over a fragment of JS using the
+            // markdown parser, which throws an error.
+            // If we can't infer the parser from from the filename, either
+            // because no filename was provided or because there is no parser
+            // found for the filename, use javascript.
+            // This is added to the options first, so that
+            // prettierRcOptions and eslintPrettierOptions can still override
+            // the parser.
+            const parserBlocklist = [null, 'graphql', 'markdown', 'html'];
```

Comment:
This list seems like it might plausibly need to expand in the future. The detailed comment above it with background is definitely appreciated, but since the situation is fairly complex and difficult to comprehend, is there a simple description we can add for what the list should contain for future reference?

For example, maybe we could add a comment stating that:

> This list should contain the list of `prettier` parser names for file types where:
>
> * Prettier supports parsing the file type
> * There is an ESLint processor that extracts JavaScript snippets from the file type.